airbyte-cdk/bulk/toolkits/load-iceberg-parquet/src/main/kotlin/io/airbyte/cdk/load/data/iceberg_parquet/AirbyteTypeToIcebergSchema.kt

```
@@ -0,0 +1,109 @@
+                                    UUID.randomUUID().hashCode(),
+                                    name,
+                                    convert(field.type)
+                                )
+                            } else {
+                                NestedField.required(
+                                    UUID.randomUUID().hashCode(),
+                                    name,
+                                    convert(field.type)
+                                )
+                            }
+                        }
+                        .toTypedArray()
+                )
+            }
+            is ArrayType -> {
+                val convert = convert(airbyteSchema.items.type)
+                if (airbyteSchema.items.nullable) {
+                    return Types.ListType.ofOptional(UUID.randomUUID().hashCode(), convert)
+                }
+                return Types.ListType.ofRequired(UUID.randomUUID().hashCode(), convert)
+            }
+            is ArrayTypeWithoutSchema ->
+                throw IllegalArgumentException("Array type without schema is not supported")
+            is BooleanType -> Types.BooleanType.get()
+            is DateType -> Types.DateType.get()
+            is IntegerType -> Types.LongType.get()
+            is NumberType -> Types.DoubleType.get()
+            is ObjectTypeWithEmptySchema ->
+                throw IllegalArgumentException("Object type with empty schema is not supported")
+            is ObjectTypeWithoutSchema ->
+                throw IllegalArgumentException("Object type without schema is not supported")
+            is StringType -> Types.StringType.get()
+            is TimeTypeWithTimezone,
+            is TimeTypeWithoutTimezone -> Types.TimeType.get()
+            is TimestampTypeWithTimezone -> Types.TimestampType.withZone()
+            is TimestampTypeWithoutTimezone -> Types.TimestampType.withoutZone()
+            is UnionType -> {
```

we should do the same thing as in parquet https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/bulk/toolkits/load-parquet/src/main/kotlin/io/airbyte/cdk/load/data/parquet/ParquetMapperPipelineFactory.kt#L27

e.g. `Union(StringType, IntType)` -> `ObjectType(mapOf("type" to StringType, "string" to StringType, "int" to IntType)` (and then `{"the_field": "foo"}` would get written out as `{"the_field": {"type": "string", "string": "foo"}}`

---

(which is to say, by the time we reach this method, we should never receive a `UnionType`, so this should throw an exception)

---

@edgao Am already using ParquetMapperPipelineFactory. It does have a condition though that if the union type only contains only 1 sub type then it will let union type be as it is.

```
  override fun mapUnion(schema: UnionType): AirbyteType {
        if (schema.options.size < 2) {
            return schema
        }
        /* 
```